lnd_test.go

```diff
@@ -13852,6 +13852,124 @@ func testChannelBackupRestore(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 }
 
+// testTLSAutoRegeneration creates an expired TLS certificate, testing that a new TLS certificate pair is regenerated when the old pair expires
+func testTLSAutoRegeneration(lnNet *lntest.NetworkHarness, t *harnessTest) {
+	certPath := lnNet.Alice.Cfg.TLSCertPath
+	keyPath := lnNet.Alice.Cfg.TLSKeyPath
+
+	// Create an expired certificate
+	expiredCert := genExpiredCertPair(t, lnNet, certPath, keyPath, time.Now())
+
+	// Restart the node to test that the test is automatically regenerated
+	lnNet.RestartNode(lnNet.Alice, nil, nil)
+
+	// Grab the newly generated certificate
+	newCertData, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		t.Fatalf("couldn't grab new certificate")
+	}
+
+	newCert, err := x509.ParseCertificate(newCertData.Certificate[0])
+	if err != nil {
+		t.Fatalf("couldn't parse new certificate")
+	}
+
+	// Test that the expired certificate was deleted and a new certificate was generated.
+	if expiredCert == newCert {
+		t.Fatalf("Expired certificate was not replaced successfully")
+	}
+
+	if !newCert.NotAfter.After(expiredCert.NotAfter) {
+		t.Fatalf("New certificate expiration is too old")
+	}
+}
+
+// genExpiredCertPair generates an expired key/cert pair to the paths provided to test that expired certificates are being regenerated correctly
+func genExpiredCertPair(t *harnessTest, lnNet *lntest.NetworkHarness, certPath string, keyPath string, expiredTime time.Time) *x509.Certificate {
```

Comment:
a couple things here:
 - `expiredTime` is not used anywhere
 - function signature should be wrapped to 80 chars
 - since `certPath` and `keyPath` are of the same type `string`, they can be defined as `certPath, keyPath string` to save space